core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -355,71 +475,90 @@ int findVersion() {
    * an attempt will be made to delete the source file.
    *
    * @param fs the filesystem used for the rename
-   * @param src the source file
-   * @param dst the destination file
+   * @param tempMetaDataFile the source file
+   * @param finalMetaDataFile the destination file
    */
-  private void renameToFinal(FileSystem fs, Path src, Path dst, int nextVersion) {
+  @VisibleForTesting
+  boolean commitNewVersion(
+      FileSystem fs, Path tempMetaDataFile, Path finalMetaDataFile, Integer nextVersion)
+      throws IOException {
     try {
-      if (!lockManager.acquire(dst.toString(), src.toString())) {
+      if (!lockManager.acquire(finalMetaDataFile.toString(), finalMetaDataFile.toString())) {
```

Comment:
Don't rename variables without a reason, you also introduced another bug here in doing so. The second string here should be tempMetadataFile

Reply:
According to the previous logic, this lock doesn't really work at all. So I did it on purpose.

Reply:
Whilst I don't think it makes sense to put the locks in this way, it really isn't relevant to the current PR and I will restore them.

Reply:
got it